_posts/2015-12-01-minutes-offline-case-study.md

```diff
@@ -0,0 +1,101 @@
+
+For now, we know our scope, we know the amounts of data our users are storing, and everything’s cool.
+
+# Offlining functionality
+
+Thanks to Hoodie, this section will be quite short. As said above, Hoodie abstracts away all the syncing and offline logic, when we use __hoodie.store.add()__ to save data, we can rely on it being stored locally and synced to the server whenever possible. If our UI only ever reacts to events that Hoodie emits, such as __hoodie.store.on(’add’, ’meeting-item’)__, and only ever fetches data through the Hoodie API, we can rely on it going through the local store first and never hitting a non-existent network.
+
+So as long as our interaction with our data store runs exclusively through Hoodie, the app is simply going to work offline with no extra effort on our part.
+
+Users have their meetings with them at all times, whether they’re on a plane, train, or submarine, and they can always take new meeting minutes, too. The one thing we can’t do offline, of course, is send the emails with the meeting minutes. For that, Hoodie has a task API:
+
+## Handling tasks that cannot possibly be completed without a network connection
+
+Some things can’t be done offline. We can’t send emails without a network connection, period. We can’t do things on the server if we can’t reach it. To help us handle cases like this, Hoodie’s task API lets us define tasks we want the server to perform, and syncs them like any other piece of data. So the task __”hey server, please email this meeting to the following addresses”__ is stored locally in the browser, and synced to the server whenever possible, which will then send the emails and sync back a task confirmation whenever possible. When the user is online, this simply sends the emails immediately, because the process is practically instantaneous, but when they’re offline, nothing breaks. It’s like any email client’s outbox.
+
+## Handling image uploads when all you’ve got is localStorage and no internet
+
+I wasn’t being entirely truthful earlier when I said that minutes.io users only store text data: they can also add custom logos to their meeting notes and emails. Of course, since we’re currently limited to localStorage, we can’t store huge images. Also, since this feature has to work offline, we can’t rely on the server to do anything for us, like image resizing. Luckily, neither of these issues is a problem.
```

Comment:
"neither of these issues is a problem" > "neither of these issues are a problem"